oat-sa/tao-item-runner-qti-fe
# Feature/ADF-1417/Upgrade to Node 18

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417) and [ADF-1419](https://oat-sa.atlassian.net/browse/ADF-1419)

Requires: 
 - [x] update the file [`package.json`](https://github.com/oat-sa/tao-item-runner-qti-fe/blob/5f48fc8b67da62353e06ad47086148f458ffab64/package.json#L52-L58) once the companion PRs have been merged and released

### Summary

Update the devDependencies for supporting Node 18

Set Node 18 for the CI workflow.

### Details

To properly support Node 18, the dev dependencies had to be updated and aligned.

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

### How to test
- check the report on GitHub
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package: `npm i`
- run the unit tests: `npm test`, `npm run test:cov`, `npm run coverage`
- test the bundling: `npm run build:all`
- test the linting: `npm run lint:src`, `npm run lint:test`


[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ADF-1419]: https://oat-sa.atlassian.net/browse/ADF-1419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
